<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>GSMR: testing the putative causal effect of Vitamin-D on about 70 traits</title>

<script src="site_libs/jquery-1.12.4/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/lumen.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<script src="site_libs/htmlwidgets-1.2/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.4/datatables.js"></script>
<link href="site_libs/dt-core-1.10.16/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.10.16/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.10.16/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.0.0/css/crosstalk.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.0.0/js/crosstalk.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />

</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 54px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 59px;
  margin-top: -59px;
}

.section h2 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h3 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h4 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h5 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h6 {
  padding-top: 59px;
  margin-top: -59px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>






<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Vitamin-D | Causality</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="index.html">Intro</a>
</li>
<li>
  <a href="1_GWAS.html">GWAS</a>
</li>
<li>
  <a href="2_GSMR.html">GSMR</a>
</li>
<li>
  <a href="3_LDSC.html">LDSC</a>
</li>
<li>
  <a href="4_mtCOJO.html">mtCOJO</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">GSMR: testing the putative causal effect of Vitamin-D on about 70 traits</h1>

</div>


<hr>
<p><em>Yan Holtz, Zhihong Zhu, Julanne Frater, Perry Bartlett, Jian Yang, John McGrath</em></p>
<hr>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<hr />
<p><a href="%22http://cnsgenomics.com/software/gsmr/%22">GSMR</a> is a tool that allows to run mendelian randomization on summary statistics of GWASs. Basically, its use will test if vitamin-D has a causal effect on other traits and diseases.</p>
<p><br></p>
<p>The GSMR pipeline requires two types of input file:</p>
<ul>
<li><p><em>Risk factor GWAS</em> - This is the GWAS computed on Vitamin-D concentration and described in the previous tab. Note that waiting for the UKB data, the data of <a href="https://www.nature.com/articles/s41467-017-02662-2">Xia et al.</a> will be used. (sample size of about 80k people).</p></li>
<li><em>Outcome GWASes</em> - GWAS summary statistics for traits potentially influenced by vitamin D concentration. Note that all outcome gwases are listed in <a href="">supplementary file</a> with their associated publication and main features. 75 traits are considered and divided in three categories.
<ul>
<li>Primary traits: 28 traits and diseases with a strong prior for vitamin D effect.</li>
<li>Secondary traits: 35 traits and diseases with no prior for vitamin D effect.</li>
<li>Risk factors: 12 traits potentially involved in the mechanism.</li>
</ul></li>
</ul>
<p><br></p>
<p><em>Note</em>: cleaning the GWAS summary statistics has been a big work. This step is explained in the script <code>cleaning_GWAS_sumstat.sh</code></p>
</div>
<div id="run-gsmr" class="section level1">
<h1>Run GSMR</h1>
<hr />
<p><u>Step 1</u>: get the last <a href="http://cnsgenomics.com/software/gcta/#Overview">GCTA version</a> that support GSMR. This program is now available on my delta folder at <code>~/bin/gcta64</code></p>
<p><br></p>
<p><u>Step 2</u>: prepare files - split to have on file per outcome - run GSMR - concatenate results together - transfer result locally. This is done on <code>delta</code>.</p>
<pre class="bash"><code># Specific repo
cd /shares/compbio/Group-Wray/YanHoltz/VITAMIND_XIA_ET_AL/3_GSMR

# Prepare a file that gives the location of every b-file (one per chromosome)
ls /gpfs/gpfs01/polaris/Q0286/UKBiobank/v2EURu_HM3/ukbEURu_imp_chr*_v2_HM3_QC.bed | sed &#39;s/.bed//&#39; &gt; gsmr_ref_data.txt

# prepare a file that gives the link to the GWAS result for the risk factor = Vitamin D
echo &quot;vitaminD /shares/compbio/Group-Wray/YanHoltz/DATA/GWAS/XiaEtAl_VitaminD/GWAS_vitaminD_XiaEtAL.ma&quot; &gt; gsmr_exposure.txt

# prepare ONE file that lists all the outcomes. This file has been build in Excel format locally
cd /Users/y.holtz/Dropbox/QBI/4_UK_BIOBANK_GWAS_PROJECT/VitaminD-Causality/0_DATA
R
library(gdata)
tmp &lt;- read.xls(&quot;list_of_traits_GSMR.xlsx&quot;, header=T)
tmp &lt;- tmp[ , c(&quot;Trait&quot;,&quot;File&quot;)]
tmp$File &lt;- paste(&quot;/shares/compbio/Group-Wray/YanHoltz/DATA/GWAS/GWAS_SUMSTAT/&quot;, tmp$File, sep=&quot;&quot;)
write.table(tmp, file=&quot;list_of_traits_GSMR.txt&quot;, row.names=FALSE, quote=FALSE, sep=&quot;\t&quot;)
q(&quot;no&quot;)
scp list_of_traits_GSMR.txt y.holtz@delta.imb.uq.edu.au:/shares/compbio/Group-Wray/YanHoltz/VITAMIND_XIA_ET_AL/3_GSMR
cat list_of_traits_GSMR.txt | sed &#39;s/ /_/g&#39; &gt; gsmr_outcome.txt

# Split this file: one file per outcome:
split -l 1 --numeric-suffixes gsmr_outcome.txt 
for i in x0* ; do a=$(echo $i | sed &#39;s/x0/x/&#39;) ; mv $i $a ; done

# send an array of GSMR
tmp_command=&quot;gcta64 --mbfile gsmr_ref_data.txt --gsmr-file gsmr_exposure.txt x{TASK_ID}  --gsmr-direction 0 --out gsmr_result_vitaminDXiaEtAl_{TASK_ID}&quot;
qsubshcom &quot;$tmp_command&quot; 1 30G GSMR_array 10:00:00 &quot;-array=1-72&quot;

# Once it&#39;s over, concatenate the results in a unique file
cat gsmr_result_vitaminDXiaEtAl_*gsmr | head -1 &gt; gsmr_result_vitaminDXiaEtAl.gsmr
cat gsmr_result_vitaminDXiaEtAl_*gsmr | grep -v &quot;Exposure&quot; &gt;&gt;  gsmr_result_vitaminDXiaEtAl.gsmr
rm *[0-9].gsmr x* *badsnps qsub* *log

# Then from locally to delta
cd /Users/y.holtz/Dropbox/QBI/4_UK_BIOBANK_GWAS_PROJECT/VitaminD-Causality/0_DATA
scp y.holtz@delta.imb.uq.edu.au:/shares/compbio/Group-Wray/YanHoltz/VITAMIND_XIA_ET_AL/3_GSMR/gsmr_result_vitaminDXiaEtAl.gsmr .</code></pre>
<p><u>Step 3</u>: read results. Description:</p>
<ul>
<li>The <strong>exposure</strong> is always the Vitamin-D, and we test its causality on several outcomes.<br />
</li>
<li><strong>Bxy</strong> is the effect of the exposure on the disease. If this effect is positive, an increase of the exposure creates a increase in the Outcome.</li>
<li><strong>se</strong> is the standard error around Bxy</li>
<li><strong>p</strong> is the pvalue of the association test. We will consider a first threshold of 0.05 for significance, and a second for multiple testing correction (0.05 / # of tests)</li>
<li><strong>nsnp</strong> is the number of SNP with a significant effect on vitamin-D. It is not always the same because a few SNP are sometime discarded by the Heidi outlier test.</li>
</ul>
<pre class="r"><code># Libraries
library(tidyverse)
library(DT)

# Read gsmr result
gsmr &lt;- read.table(&quot;0_DATA/gsmr_result_vitaminDXiaEtAl.gsmr&quot;, header = T) %&gt;%
  mutate(Outcome=gsub(&quot;_&quot;, &quot; &quot;, Outcome))

# Read the meaning of files:
library(gdata)
meaning &lt;- read.xls(&quot;0_DATA/list_of_traits_GSMR.xlsx&quot;, header=T)
primary &lt;- gsmr %&gt;% filter( Outcome %in% meaning$Trait[ which(meaning$Group==&quot;Primary&quot;) ] )
secondary &lt;- gsmr %&gt;% filter( Outcome %in% meaning$Trait[ which(meaning$Group==&quot;Secondary&quot;) ] )
risk &lt;- gsmr %&gt;% filter( Outcome %in% meaning$Trait[ which(meaning$Group==&quot;Risk&quot;) ] )

# Compute thresholds
thres_primary &lt;- 0.05 / nrow(primary)
thres_secondary &lt;- 0.05 / nrow(secondary)
thres_risk &lt;- 0.05 / nrow(risk)</code></pre>
</div>
<div id="result-primary-traits" class="section level1 tabset">
<h1>Result: primary traits</h1>
<hr />
<p>In total 27 primary traits were studied. The threshold used to take multiple testing into account is 0.05 / # of test. Here is a description of the effect size and pvalues obtained with the GSMR test:</p>
<div id="effect-size" class="section level2">
<h2>Effect size</h2>
<p>The following chart describes the <code>effect sizes</code> (bxy) of vitamin D on each primary traits. The standard error is indicated in green.</p>
<pre class="r"><code>#plot
primary %&gt;%
  arrange(bxy) %&gt;%
  filter(!is.na(bxy) ) %&gt;% 
  mutate(name=factor(Outcome, unique(Outcome))) %&gt;%
  mutate(significance=ifelse(p&lt;thres_primary, &quot;signif multi.testing&quot;, ifelse(p&lt;0.05, &quot;signif&quot;, &quot;nonSignif&quot;) ) ) %&gt;%
  ggplot( aes(x=name, y=bxy)) +
    geom_hline( yintercept=0 ) +
    geom_segment( aes(x=name, xend=name, y=bxy-se, yend=bxy+se), color=&quot;grey&quot;, alpha=0.7) +
    geom_point(aes(color=significance), size=4) +
    scale_color_manual( values=c(&quot;grey&quot;, &quot;orange&quot;, &quot;red&quot;)) +
    coord_flip() +
    theme_bw() +
    theme( panel.grid.major.y = element_line(size=0.1)) +
    ylab(&quot;Bxy (Effect size)&quot;) +
    xlab(&quot;&quot;)</code></pre>
<div class="figure" style="text-align: center">
<img src="2_GSMR_files/figure-html/unnamed-chunk-3-1.png" alt="Figure: effect size (Bxy) of Vitamin D on all primary traits. Grey / orange / red points show significance of the causality. Blue lines show the standard error around the effect size." width="768" />
<p class="caption">
Figure: effect size (Bxy) of Vitamin D on all primary traits. Grey / orange / red points show significance of the causality. Blue lines show the standard error around the effect size.
</p>
</div>
<pre class="r"><code>#Save at png
ggsave(filename = &quot;IMG/gsmrPrimaryTraits.png&quot;)</code></pre>
<p><br></p>
<p>A few observation:</p>
<ul>
<li>most of the effect size are negative: in these cases a lack of vitamin D increase the chances to develop a disease.</li>
<li>few effect are significant. This is probably due to the fact that the number of genetic instrument is currently very low (6 significant SNPs only)</li>
<li>it is interesting to note the strong association between vitamin D and <code>Alzheimer's disease</code>. Two GWAS are available for this disease and both give a significant result.</li>
</ul>
</div>
<div id="pvalues" class="section level2">
<h2>Pvalues</h2>
<p>Here is an ordered barplot of the pvalues. We have several traits over the 0.05 threshold of significance. But this does not account for multiple testing.</p>
<pre class="r"><code>primary %&gt;% 
  arrange(desc(p)) %&gt;%
  filter( !is.na(bxy)) %&gt;%
  mutate(name=factor(Outcome, unique(Outcome))) %&gt;%
  mutate(significance=ifelse(p&lt;thres_primary, &quot;signif multi.testing&quot;, ifelse(p&lt;0.05, &quot;signif&quot;, &quot;nonSignif&quot;) ) ) %&gt;%
  ggplot( aes(x=name, y=-log10(p))) +
    geom_segment( aes(x=name, xend=name, y=-log10(p), yend=0), color=&quot;grey&quot;, alpha=0.7) +
    geom_point(aes(color=significance), size=4) +
    scale_color_manual( values=c(&quot;grey&quot;, &quot;orange&quot;, &quot;red&quot;)) +
    coord_flip() +
    theme_light() +
    theme( panel.grid.major.y = element_blank()) +
    ylab(&quot;-log10( pvalue )&quot;) +
    xlab(&quot;&quot;) </code></pre>
<p><img src="2_GSMR_files/figure-html/unnamed-chunk-4-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="result-secondary-traits" class="section level1 tabset">
<h1>Result: secondary traits</h1>
<hr />
<p>In total 27 primary traits were studied. The threshold used to take multiple testing into account is 0.05 / # of test. Here is a description of the effect size and pvalues obtained with the GSMR test:</p>
<div id="effect-size-1" class="section level2">
<h2>Effect size</h2>
<pre class="r"><code># Plot
secondary %&gt;%
  arrange(bxy) %&gt;%
  filter(!is.na(bxy) ) %&gt;% 
  mutate(name=factor(Outcome, unique(Outcome))) %&gt;%
  mutate(significance=ifelse(p&lt;thres_secondary, &quot;signif multi.testing&quot;, ifelse(p&lt;0.05, &quot;signif&quot;, &quot;nonSignif&quot;) ) ) %&gt;%
  ggplot( aes(x=name, y=bxy)) +
    geom_hline( yintercept=0 ) +
    geom_segment( aes(x=name, xend=name, y=bxy-se, yend=bxy+se), color=&quot;grey&quot;, alpha=0.7) +
    geom_point(aes(color=significance), size=4) +
    scale_color_manual( values=c(&quot;grey&quot;, &quot;orange&quot;, &quot;red&quot;)) +
    coord_flip() +
    theme_bw() +
    theme( panel.grid.major.y = element_line(size=0.1)) +
    ylab(&quot;Bxy (Effect size)&quot;) +
    xlab(&quot;&quot;)</code></pre>
<div class="figure" style="text-align: center">
<img src="2_GSMR_files/figure-html/unnamed-chunk-5-1.png" alt="Figure: effect size (Bxy) of Vitamin D on all primary traits. Grey / orange / red points show significance of the causality. Blue lines show the standard error around the effect size." width="768" />
<p class="caption">
Figure: effect size (Bxy) of Vitamin D on all primary traits. Grey / orange / red points show significance of the causality. Blue lines show the standard error around the effect size.
</p>
</div>
<pre class="r"><code>#Save at png
ggsave(filename = &quot;IMG/gsmrSecondaryTraits.png&quot;)</code></pre>
<p>A few observation:</p>
<ul>
<li>2 significant associations after multiple testing correction.</li>
<li>2 of have a positive effect size: more Vitamin-D, more trait.</li>
<li>0 have a negative effect size.</li>
</ul>
</div>
<div id="pvalues-1" class="section level2">
<h2>Pvalues</h2>
<p>Here is an ordered barplot of the pvalues. We have several traits over the 0.05 threshold of significance. But this does not account for multiple testing.</p>
<pre class="r"><code>secondary %&gt;% 
  arrange(desc(p)) %&gt;%
  filter( !is.na(bxy)) %&gt;%
  mutate(name=factor(Outcome, unique(Outcome))) %&gt;%
  mutate(significance=ifelse(p&lt;thres_secondary, &quot;signif multi.testing&quot;, ifelse(p&lt;0.05, &quot;signif&quot;, &quot;nonSignif&quot;) ) ) %&gt;%
  ggplot( aes(x=name, y=-log10(p))) +
    geom_segment( aes(x=name, xend=name, y=-log10(p), yend=0), color=&quot;grey&quot;, alpha=0.7) +
    geom_point(aes(color=significance), size=4) +
    scale_color_manual( values=c(&quot;grey&quot;, &quot;orange&quot;, &quot;red&quot;)) +
    coord_flip() +
    theme_light() +
    theme( panel.grid.major.y = element_blank()) +
    ylab(&quot;-log10( pvalue )&quot;) +
    xlab(&quot;&quot;) </code></pre>
<p><img src="2_GSMR_files/figure-html/unnamed-chunk-6-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="result-risk-factors" class="section level1 tabset">
<h1>Result: Risk factors</h1>
<hr />
<div id="effect-size-2" class="section level2">
<h2>Effect size</h2>
<p>A few observation:</p>
<ul>
<li>1 significant associations after multiple testing correction.</li>
<li>1 of have a positive effect size: more Vitamin-D, more trait.</li>
<li>0 have a negative effect size.</li>
</ul>
<pre class="r"><code>#plot
risk %&gt;%
  arrange(bxy) %&gt;%
  filter(!is.na(bxy) ) %&gt;% 
  mutate(name=factor(Outcome, unique(Outcome))) %&gt;%
  mutate(significance=ifelse(p&lt;thres_risk, &quot;signif multi.testing&quot;, ifelse(p&lt;0.05, &quot;signif&quot;, &quot;nonSignif&quot;) ) ) %&gt;%
  ggplot( aes(x=name, y=bxy)) +
    geom_hline( yintercept=0 ) +
    geom_segment( aes(x=name, xend=name, y=bxy-se, yend=bxy+se), color=&quot;skyblue&quot;, alpha=0.7) +
    geom_point(aes(color=significance), size=4) +
    scale_color_manual( values=c(&quot;grey&quot;, &quot;orange&quot;, &quot;red&quot;)) +
    coord_flip() +
    theme_bw() +
    theme( panel.grid.major.y = element_line(size=0.1)) +
    ylab(&quot;Bxy (Effect size)&quot;) +
    xlab(&quot;&quot;)</code></pre>
<div class="figure" style="text-align: center">
<img src="2_GSMR_files/figure-html/unnamed-chunk-7-1.png" alt="Figure: effect size (Bxy) of Vitamin D on all primary traits. Grey / orange / red points show significance of the causality. Blue lines show the standard error around the effect size." width="768" />
<p class="caption">
Figure: effect size (Bxy) of Vitamin D on all primary traits. Grey / orange / red points show significance of the causality. Blue lines show the standard error around the effect size.
</p>
</div>
<pre class="r"><code>#Save at png
ggsave(filename = &quot;IMG/gsmrRiskFactors.png&quot;)</code></pre>
</div>
<div id="pvalues-2" class="section level2">
<h2>Pvalues</h2>
<p>Here is an ordered barplot of the pvalues. We have several traits over the 0.05 threshold of significance. But this does not account for multiple testing.</p>
<pre class="r"><code>risk %&gt;% 
  arrange(desc(p)) %&gt;%
  filter( !is.na(bxy)) %&gt;%
  mutate(name=factor(Outcome, unique(Outcome))) %&gt;%
  mutate(significance=ifelse(p&lt;thres_risk, &quot;signif multi.testing&quot;, ifelse(p&lt;0.05, &quot;signif&quot;, &quot;nonSignif&quot;) ) ) %&gt;%
  ggplot( aes(x=name, y=-log10(p))) +
    geom_segment( aes(x=name, xend=name, y=-log10(p), yend=0), color=&quot;skyblue&quot;, alpha=0.7) +
    geom_point(aes(color=significance), size=4) +
    scale_color_manual( values=c(&quot;grey&quot;, &quot;orange&quot;, &quot;red&quot;)) +
    coord_flip() +
    theme_light() +
    theme( panel.grid.major.y = element_blank()) +
    ylab(&quot;-log10( pvalue )&quot;) +
    xlab(&quot;&quot;) </code></pre>
<p><img src="2_GSMR_files/figure-html/unnamed-chunk-8-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="table" class="section level1">
<h1>Table</h1>
<hr />
<p>The 71 GSMR test results are displayed in this table in case you need to check a specific value:</p>
<pre class="r"><code>datatable(gsmr %&gt;% arrange(p), rownames = FALSE, options = list(pageLength = 10) )</code></pre>
<div id="htmlwidget-2bd019e50a1c1755f992" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-2bd019e50a1c1755f992">{"x":{"filter":"none","data":[["vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD","vitaminD"],["Myeloid","Granulocyte count","Neutrophil","Height","Alzheimer's Disease","Basophil count","Platelet Count","Lymphocyte count","HDL cholesterol","allergic rhinitis","Plateletcrit","Monocyte count","Dementia - Alzheimer","hypertensive disease","insomnia","Mean Corpuscular Hemoglobin","Macular degeneration","Osteoarthritis","Iron deficiency anemias","Systolic blood pressure","acute reaction to stress","Reticulocyte count","Platelet ditribution width","Eosinophilic esophagitis","IQ (UKB)","hernia abdominopelvic cavity","BMI","Depressive Symptoms","Ulcerative Colitis","psychiatric disorder","Major depressive disorder","Coronary Artery Disease","Hemoglobin","Autism Spectrum Disorder","dermatophytosis","hemorrhoids","Major Depressive Disorder","Red cell distribution width","Dyslipidemia","Type II diabetes","Peripheral vascular disease","Mean platelet volume ","Neuroticism","Diastolic blood pressure","IQ","Schizophrenia","Tanning response to sun exposure","Peptic ulcers","Rheumatoid Arthritis","Parkinson's Disease","Stroke","Aged Macular Degeneration","cardiovascular disease","Irritable bowel syndrome","Outdoor activity (IPAQ)","Waist-to-hip ratio adjusted for BMI","Asthma","Crohn's Disease","Bipolar Disorder","Disease count","Hematocrit","LDL cholesterol","Hippocampal volume","Red blood cell count","Mean corpuscular volume","osteoporosis","Edu Year","LDL cholesterol","Cancer","Smoking","Immature Fraction of Reticulocytes"],[0.188634,0.183935,0.1788,0.11996,-0.471287,0.0956538,0.0974073,0.0916145,0.120789,0.273004,0.0871263,0.0820581,-0.156573,0.174548,-0.410881,-0.0695121,0.451325,0.157792,-0.363941,-0.0610669,0.285096,-0.0453709,-0.0433942,0.0430098,0.0382714,0.146085,0.0390816,0.0365254,-0.217529,-0.139598,-0.120643,-0.0903309,-0.0354376,-0.270037,0.172684,-0.11376,-0.198607,0.0323647,-0.078836,-0.119132,0.166394,-0.0281917,0.0262072,-0.0338941,-0.0391354,-0.0753773,-0.0548805,0.166237,-0.106079,-0.127228,-0.0451323,0.0855171,0.0548223,0.0986047,-0.00825271,0.0137426,-0.024106,-0.0642234,0.0591033,0.0145178,0.00865116,0.0116192,0.0126072,0.00387763,-0.00294016,-0.0133888,-0.00177883,-0.00274909,0.00427858,0.00165813,0.0012178],[0.0383439,0.0382306,0.0381451,0.0304114,0.164363,0.0369583,0.0385448,0.0380294,0.0507475,0.116511,0.0386905,0.0377571,0.0732505,0.0872544,0.21424,0.0373633,0.255748,0.103674,0.245607,0.0467072,0.230119,0.0379479,0.0379387,0.0376307,0.0353065,0.138459,0.0384971,0.0367666,0.219458,0.142993,0.124603,0.0951306,0.03746,0.285764,0.185889,0.126399,0.222408,0.0373557,0.0913678,0.141191,0.203082,0.0381701,0.0358304,0.047101,0.054881,0.10909,0.0884845,0.270175,0.182646,0.219072,0.0832583,0.158345,0.103689,0.192739,0.0216492,0.0437958,0.0875665,0.238288,0.221769,0.0546114,0.0372754,0.0556041,0.0897973,0.0375836,0.0372534,0.190097,0.0279101,0.0493619,0.104364,0.0424752,0.0374413],[8.67511e-07,1.50021e-06,2.76738e-06,7.99435e-05,0.00413909,0.00964912,0.0115003,0.0159943,0.0173034,0.0191215,0.0243299,0.0297567,0.0325573,0.0454511,0.0551293,0.0628236,0.0776103,0.128008,0.138392,0.191063,0.215381,0.231848,0.252708,0.253061,0.278375,0.29139,0.310018,0.320495,0.321583,0.328935,0.332932,0.342343,0.344142,0.344677,0.35291,0.368113,0.371865,0.386276,0.388224,0.3988,0.41259,0.460161,0.464521,0.471768,0.475786,0.489589,0.535108,0.538362,0.561381,0.561404,0.587766,0.58915,0.597001,0.608934,0.703053,0.753682,0.783095,0.787529,0.789848,0.790363,0.81647,0.834478,0.888347,0.917825,0.937094,0.94385,0.949182,0.955587,0.967298,0.96886,0.974053],[17,17,17,15,17,17,17,17,16,17,17,16,17,16,17,17,17,17,17,17,17,17,17,17,17,17,17,16,17,17,17,17,17,17,17,17,17,17,17,16,17,17,14,15,17,17,16,17,17,17,17,16,17,17,17,17,17,16,16,17,17,16,17,17,17,17,17,16,17,17,17]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Exposure<\/th>\n      <th>Outcome<\/th>\n      <th>bxy<\/th>\n      <th>se<\/th>\n      <th>p<\/th>\n      <th>nsnp<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[2,3,4,5]}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="reverse-gsmr" class="section level1">
<h1>Reverse GSMR</h1>
<hr />
<p>Work in progress</p>
<pre class="r"><code># Specific repo
cd /shares/compbio/Group-Wray/YanHoltz/VITAMIND_XIA_ET_AL/3_GSMR/REVERSE

# Prepare a file that gives the location of every b-file (one per chromosome) --&gt; I don&#39;t take hapmap3, cauz may be the signif SNP are not in Hapmap3
ls /gpfs/gpfs01/polaris/Q0286/UKBiobank/v2EUR_geno/ukbEUR_cal_chr*_v2_QC.bed | sed &#39;s/.bed//&#39; &gt; gsmr_ref_data.txt

# copy GWAS sumstat files done for the forward GSMR
cp ../gsmr_exposure.txt ../gsmr_outcome.txt .

# Split this file: one file per outcome:
split -l 1 --numeric-suffixes gsmr_outcome.txt 
for i in x0* ; do a=$(echo $i | sed &#39;s/x0/x/&#39;) ; mv $i $a ; done

# send an array of GSMR
tmp_command=&quot;gcta64 --mbfile gsmr_ref_data.txt --gsmr-file gsmr_exposure.txt x{TASK_ID}  --gsmr-direction 1 --out gsmr_reverse_result_vitaminDXiaEtAl_{TASK_ID}&quot;
qsubshcom &quot;$tmp_command&quot; 1 30G GSMR_array 10:00:00 &quot;-array=1-72&quot;

# Once it&#39;s over, concatenate the results in a unique file
cat gsmr_reverse_result*gsmr | head -1 &gt; gsmr_reverse_result_vitaminDXiaEtAl.gsmr
cat gsmr_reverse_result*gsmr | grep -v &quot;Exposure&quot; &gt;&gt;  gsmr_reverse_result_vitaminDXiaEtAl.gsmr
rm *[0-9].gsmr x* *badsnps qsub*</code></pre>
</div>
<div id="conclusion" class="section level1">
<h1>- Conclusion</h1>
<hr />
<style>
div.blue { background-color:#e6f0ff; border-radius: 5px; padding: 20px;}
</style>
<div class="blue">
<ul>
<li>GSMR analysis suggests that vitamin-D concentration has a causal effect on several traits like Hypertension or Alzheimer. This sould be verified using a bigger sample size and multiple testing correction.</li>
</ul>
</div>
</div>

<br><br>
<hr />
<p style="text-align: center; font-size: 80%">A work by <a href="https://github.com/holtzy/">Yan Holtz</a></p>
<p style="text-align: center; font-size: 80%"><span style="color: #808080;"><em>Yan.holtz.data@gmail.com</em></span></p>

<!-- Add icon library -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">

<!-- Add font awesome icons -->
<p style="text-align: center;">
	<a href="https://twitter.com/r_graph_gallery?lang=en" class="fa fa-twitter"></a>
	<a href="https://www.linkedin.com/in/yan-holtz-2477534a/" class="fa fa-linkedin"></a>
	<a href="https://github.com/holtzy/" class="fa fa-github"></a>
</p>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
