<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>mtCOJO: testing the potential implication of confounding risk factors</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/lumen.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<script src="site_libs/htmlwidgets-1.2/htmlwidgets.js"></script>
<script src="site_libs/plotly-binding-4.7.1/plotly.js"></script>
<script src="site_libs/typedarray-0.1/typedarray.min.js"></script>
<link href="site_libs/crosstalk-1.0.0/css/crosstalk.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.0.0/js/crosstalk.min.js"></script>
<link href="site_libs/plotlyjs-1.29.2/plotly-htmlwidgets.css" rel="stylesheet" />
<script src="site_libs/plotlyjs-1.29.2/plotly-latest.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />

</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 54px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 59px;
  margin-top: -59px;
}

.section h2 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h3 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h4 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h5 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h6 {
  padding-top: 59px;
  margin-top: -59px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>






<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Vitamin-D | Causality</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="index.html">Intro</a>
</li>
<li>
  <a href="1_GWAS.html">GWAS</a>
</li>
<li>
  <a href="2_GSMR.html">GSMR</a>
</li>
<li>
  <a href="3_LDSC.html">LDSC</a>
</li>
<li>
  <a href="4_mtCOJO.html">mtCOJO</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">mtCOJO: testing the potential implication of confounding risk factors</h1>

</div>


<hr>
<p><em>Yan Holtz, Zhihong Zhu, Julanne Frater, Perry Bartlett, Jian Yang, John McGrath</em></p>
<hr>
<p><br><br></p>
<div id="material" class="section level1">
<h1>Material</h1>
<hr />
<p><a href="http://cnsgenomics.com/software/gcta/#mtCOJO">mtCOJO</a> allows to build a new GWAS summary statistics for a trait using summary statistics of one or several other risk factors. Here are the risk factor we considered in our study:</p>
<ul>
<li>BMI</li>
<li>Diastolic blood pressure</li>
<li>Number of years of education</li>
<li>HDL cholesterol</li>
<li>Height</li>
<li>LDL cholesterol</li>
<li>Outdoor activity</li>
<li>Smoking</li>
<li>Systolic blood pressure</li>
<li>Tanning response</li>
<li>Waist to hip ratio</li>
</ul>
<p>We considered these risk factors <code>one by one</code> in a first step, and then considered them <code>jointly</code>.</p>
<p><em>Note</em>: complete list of these risk factors with source and details is available <a href="">here</a></p>
</div>
<div id="method" class="section level1">
<h1>Method</h1>
<hr />
<p>The input of mtCOJO are:</p>
<ul>
<li><strong>mbfile</strong>: individual level genotypes of a reference sample. We will use the Hapmap3 SNPs of the UKB population. This is used for LD estimation.</li>
<li><strong>mtcojo-file</strong>: a list of the GWAS summary data and prevalence of diseases. first row is the target trait. Other rows are covariate.</li>
<li><strong>LD score references</strong>: come from the <a href="https://github.com/bulik/ldsc">LDSC tool</a>.</li>
</ul>
<p>mtCOJO will provide a new GWAS summary statistics file per risk factor. GSMR will then be run on these new files.</p>
</div>
<div id="run-mtcojo" class="section level1">
<h1>Run mtCOJO</h1>
<hr />
<p>This script run <code>mtCOJO</code> using the vitamin-D GWAS and each risk factor GWAS one by one.</p>
<pre class="r"><code># Specific repo
cd /shares/compbio/Group-Wray/YanHoltz/VITAMIND_XIA_ET_AL/8_MTCOJO

# Build mbfile
ls  /gpfs/gpfs01/polaris/Q0286/UKBiobank/v2EURu_HM3/ukbEURu_imp_chr*bed | sed &#39;s/.bed//&#39; &gt; mtcojo_mbfile.txt

# Build mtcojo file. One per risk factor, then one for all.

# BMI
echo &quot;vitaminD /shares/compbio/Group-Wray/YanHoltz/DATA/GWAS/XiaEtAl_VitaminD/GWAS_vitaminD_XiaEtAL.ma&quot; &gt; list_of_trait_BMI.txt
echo &quot;BMI /shares/compbio/Group-Wray/YanHoltz/DATA/GWAS/GWAS_SUMSTAT/bmi_giant_2015.txt&quot; &gt;&gt; list_of_trait_BMI.txt
tmp_command=&quot;gcta64 --mbfile mtcojo_mbfile.txt --mtcojo-file list_of_trait_BMI.txt --ref-ld-chr eur_w_ld_chr/ --w-ld-chr eur_w_ld_chr/ --out mtcojo_BMI_result_vitaminD&quot;
qsubshcom &quot;$tmp_command&quot; 1 30G mtCOJO 10:00:00 &quot;&quot;

# DIASTOLIC BLOOD PRESSURE
echo &quot;vitaminD /shares/compbio/Group-Wray/YanHoltz/DATA/GWAS/XiaEtAl_VitaminD/GWAS_vitaminD_XiaEtAL.ma&quot; &gt; list_of_trait_DBP.txt
echo &quot;DBP /shares/compbio/Group-Wray/YanHoltz/DATA/GWAS/GWAS_SUMSTAT/dbp_ukb_v1_2016.txt&quot; &gt;&gt; list_of_trait_DBP.txt
tmp_command=&quot;gcta64 --mbfile mtcojo_mbfile.txt --mtcojo-file list_of_trait_DBP.txt --ref-ld-chr eur_w_ld_chr/ --w-ld-chr eur_w_ld_chr/ --out mtcojo_DBP_result_vitaminD&quot;
qsubshcom &quot;$tmp_command&quot; 1 30G mtCOJO 10:00:00 &quot;&quot;

# EDU YEAR
echo &quot;vitaminD /shares/compbio/Group-Wray/YanHoltz/DATA/GWAS/XiaEtAl_VitaminD/GWAS_vitaminD_XiaEtAL.ma&quot; &gt; list_of_trait_EY.txt
echo &quot;EY /shares/compbio/Group-Wray/YanHoltz/DATA/GWAS/GWAS_SUMSTAT/eduyears_ssgac_2016.txt&quot; &gt;&gt; list_of_trait_EY.txt
tmp_command=&quot;gcta64 --mbfile mtcojo_mbfile.txt --mtcojo-file list_of_trait_EY.txt --ref-ld-chr eur_w_ld_chr/ --w-ld-chr eur_w_ld_chr/ --out mtcojo_EY_result_vitaminD&quot;
qsubshcom &quot;$tmp_command&quot; 1 30G mtCOJO 10:00:00 &quot;&quot;

# HDL Cholesterol
echo &quot;vitaminD /shares/compbio/Group-Wray/YanHoltz/DATA/GWAS/XiaEtAl_VitaminD/GWAS_vitaminD_XiaEtAL.ma&quot; &gt; list_of_trait_HDL.txt
echo &quot;HDL /shares/compbio/Group-Wray/YanHoltz/DATA/GWAS/GWAS_SUMSTAT/hdl_glgc_2013.txt&quot; &gt;&gt; list_of_trait_HDL.txt
tmp_command=&quot;gcta64 --mbfile mtcojo_mbfile.txt --mtcojo-file list_of_trait_HDL.txt --ref-ld-chr eur_w_ld_chr/ --w-ld-chr eur_w_ld_chr/ --out mtcojo_HDL_result_vitaminD&quot;
qsubshcom &quot;$tmp_command&quot; 1 30G mtCOJO 10:00:00 &quot;&quot;

# HEIGHT
echo &quot;vitaminD /shares/compbio/Group-Wray/YanHoltz/DATA/GWAS/XiaEtAl_VitaminD/GWAS_vitaminD_XiaEtAL.ma&quot; &gt; list_of_trait_height.txt
echo &quot;height /shares/compbio/Group-Wray/YanHoltz/DATA/GWAS/GWAS_SUMSTAT/height_giant_2014.txt&quot; &gt;&gt; list_of_trait_height.txt
tmp_command=&quot;gcta64 --mbfile mtcojo_mbfile.txt --mtcojo-file list_of_trait_height.txt --ref-ld-chr eur_w_ld_chr/ --w-ld-chr eur_w_ld_chr/ --out mtcojo_height_result_vitaminD&quot;
qsubshcom &quot;$tmp_command&quot; 1 30G mtCOJO 10:00:00 &quot;&quot;

# LDL CHOLESTEROL
echo &quot;vitaminD /shares/compbio/Group-Wray/YanHoltz/DATA/GWAS/XiaEtAl_VitaminD/GWAS_vitaminD_XiaEtAL.ma&quot; &gt; list_of_trait_LDL.txt
echo &quot;LDL /shares/compbio/Group-Wray/YanHoltz/DATA/GWAS/GWAS_SUMSTAT/ldl_glgc_2013.txt&quot; &gt;&gt; list_of_trait_LDL.txt
tmp_command=&quot;gcta64 --mbfile mtcojo_mbfile.txt --mtcojo-file list_of_trait_LDL.txt --ref-ld-chr eur_w_ld_chr/ --w-ld-chr eur_w_ld_chr/ --out mtcojo_LDL_result_vitaminD&quot;
qsubshcom &quot;$tmp_command&quot; 1 30G mtCOJO 10:00:00 &quot;&quot;

# OUTDOOR ACTIVITY (IPAQ)
echo &quot;vitaminD /shares/compbio/Group-Wray/YanHoltz/DATA/GWAS/XiaEtAl_VitaminD/GWAS_vitaminD_XiaEtAL.ma&quot; &gt; list_of_trait_IPAQ.txt
echo &quot;IPAQ /shares/compbio/Group-Wray/YanHoltz/DATA/GWAS/GWAS_SUMSTAT/ukbEUR_IPAQG_cojo.txt&quot; &gt;&gt; list_of_trait_IPAQ.txt
tmp_command=&quot;gcta64 --mbfile mtcojo_mbfile.txt --mtcojo-file list_of_trait_IPAQ.txt --ref-ld-chr eur_w_ld_chr/ --w-ld-chr eur_w_ld_chr/ --out mtcojo_IPAQ_result_vitaminD&quot;
qsubshcom &quot;$tmp_command&quot; 1 30G mtCOJO 10:00:00 &quot;&quot;

# SMOKE
echo &quot;vitaminD /shares/compbio/Group-Wray/YanHoltz/DATA/GWAS/XiaEtAl_VitaminD/GWAS_vitaminD_XiaEtAL.ma&quot; &gt; list_of_trait_SMOKE.txt
echo &quot;SMOKE /shares/compbio/Group-Wray/YanHoltz/DATA/GWAS/GWAS_SUMSTAT/ukbEUR_SI_cojo.txt&quot; &gt;&gt; list_of_trait_SMOKE.txt
tmp_command=&quot;gcta64 --mbfile mtcojo_mbfile.txt --mtcojo-file list_of_trait_SMOKE.txt --ref-ld-chr eur_w_ld_chr/ --w-ld-chr eur_w_ld_chr/ --out mtcojo_SMOKE_result_vitaminD&quot;
qsubshcom &quot;$tmp_command&quot; 1 30G mtCOJO 10:00:00 &quot;&quot;

# SYSTOLIC BLOOD PRESSURE (SBP)
echo &quot;vitaminD /shares/compbio/Group-Wray/YanHoltz/DATA/GWAS/XiaEtAl_VitaminD/GWAS_vitaminD_XiaEtAL.ma&quot; &gt; list_of_trait_SBP.txt
echo &quot;SBP /shares/compbio/Group-Wray/YanHoltz/DATA/GWAS/GWAS_SUMSTAT/sbp_ukb_v1_2016.txt&quot; &gt;&gt; list_of_trait_SBP.txt
tmp_command=&quot;gcta64 --mbfile mtcojo_mbfile.txt --mtcojo-file list_of_trait_SBP.txt --ref-ld-chr eur_w_ld_chr/ --w-ld-chr eur_w_ld_chr/ --out mtcojo_SBP_result_vitaminD&quot;
qsubshcom &quot;$tmp_command&quot; 1 30G mtCOJO 10:00:00 &quot;&quot;

# TANNING RESPONSE (TR)
echo &quot;vitaminD /shares/compbio/Group-Wray/YanHoltz/DATA/GWAS/XiaEtAl_VitaminD/GWAS_vitaminD_XiaEtAL.ma&quot; &gt; list_of_trait_TR.txt
echo &quot;TR /shares/compbio/Group-Wray/YanHoltz/DATA/GWAS/GWAS_SUMSTAT/GWAS_tanning_UKBB.ma&quot; &gt;&gt; list_of_trait_TR.txt
tmp_command=&quot;gcta64 --mbfile mtcojo_mbfile.txt --mtcojo-file list_of_trait_TR.txt --ref-ld-chr eur_w_ld_chr/ --w-ld-chr eur_w_ld_chr/ --out mtcojo_TR_result_vitaminD&quot;
qsubshcom &quot;$tmp_command&quot; 1 30G mtCOJO 10:00:00 &quot;&quot;

# WAIST TO HIP RATIO ADJUSTED
echo &quot;vitaminD /shares/compbio/Group-Wray/YanHoltz/DATA/GWAS/XiaEtAl_VitaminD/GWAS_vitaminD_XiaEtAL.ma&quot; &gt; list_of_trait_WHR.txt
echo &quot;WHR /shares/compbio/Group-Wray/YanHoltz/DATA/GWAS/GWAS_SUMSTAT/whradjbmi_giant_2015.txt&quot; &gt;&gt; list_of_trait_WHR.txt
tmp_command=&quot;gcta64 --mbfile mtcojo_mbfile.txt --mtcojo-file list_of_trait_WHR.txt --ref-ld-chr eur_w_ld_chr/ --w-ld-chr eur_w_ld_chr/ --out mtcojo_WHR_result_vitaminD&quot;
qsubshcom &quot;$tmp_command&quot; 1 30G mtCOJO 10:00:00 &quot;&quot;

# ALL
echo &quot;vitaminD /shares/compbio/Group-Wray/YanHoltz/DATA/GWAS/XiaEtAl_VitaminD/GWAS_vitaminD_XiaEtAL.ma&quot; &gt; list_of_trait_ALL.txt
echo &quot;BMI /shares/compbio/Group-Wray/YanHoltz/DATA/GWAS/GWAS_SUMSTAT/height_giant_2014.txt&quot; &gt;&gt; list_of_trait_ALL.txt
echo &quot;SMOKE /shares/compbio/Group-Wray/YanHoltz/DATA/GWAS/GWAS_SUMSTAT/ukbEUR_SI_cojo.txt&quot; &gt;&gt; list_of_trait_ALL.txt
tmp_command=&quot;gcta64 --mbfile mtcojo_mbfile.txt --mtcojo-file list_of_trait_ALL.txt --ref-ld-chr eur_w_ld_chr/ --w-ld-chr eur_w_ld_chr/ --out mtcojo_ALL_result_vitaminD&quot;
qsubshcom &quot;$tmp_command&quot; 1 30G mtCOJO 10:00:00 &quot;&quot;</code></pre>
<p>I now have a several files called <code>mtcojo_*_result_vitaminD_result.mtcojo.cma</code> that are a GWAS summary statistics for Vitamin-D corrected for all my Risk factors.</p>
<p>Clean the folder to keep only these files:</p>
<pre class="r"><code># Clean
rm -r job_reports  list_of_trait*  *log *badsnps qsub.2018.06.25.log mtcojo_mbfile.txt</code></pre>
</div>
<div id="run-gsmr" class="section level1">
<h1>Run GSMR</h1>
<hr />
<p>I can run GSMR on this new GWAS summary statistics:</p>
<pre class="r"><code># Loop on every new sumstat to send GSMR on them
for myRiskFact in  $(ls mtcojo_*_result_vitaminD.mtcojo.cma | sed &#39;s/.*cojo_//&#39; | sed &#39;s/_.*//&#39;) ; do
  
  # Echo
  echo $myRiskFact ;

  # Specific repo
  cd /shares/compbio/Group-Wray/YanHoltz/VITAMIND_XIA_ET_AL/8_MTCOJO/
  mkdir GSMR_$myRiskFact
  cd /shares/compbio/Group-Wray/YanHoltz/VITAMIND_XIA_ET_AL/8_MTCOJO/GSMR_$myRiskFact
  
  # Prepare a file that gives the location of every b-file (one per chromosome)
  ls /gpfs/gpfs01/polaris/Q0286/UKBiobank/v2EURu_HM3/ukbEURu_imp_chr*_v2_HM3_QC.bed | sed &#39;s/.bed//&#39; &gt; gsmr_ref_data.txt
  
  # prepare a file that gives the link to the GWAS result for the risk factor = Vitamin D. After correction for risk factors
  cat /shares/compbio/Group-Wray/YanHoltz/VITAMIND_XIA_ET_AL/8_MTCOJO/mtcojo_${myRiskFact}_result_vitaminD.mtcojo.cma | cut -f1-8 &gt; file.ma
  a=$(pwd)
  echo &quot;vitaminD $a/file.ma&quot; &gt; gsmr_exposure.txt
  
  # prepare ONE file that lists all the outcomes. This file has been build in Excel format locally
  cp /shares/compbio/Group-Wray/YanHoltz/VITAMIND_XIA_ET_AL/3_GSMR/gsmr_outcome.txt .
  
  # Split this file: one file per outcome:
  split -l 1 --numeric-suffixes gsmr_outcome.txt 
  for i in x0* ; do a=$(echo $i | sed &#39;s/x0/x/&#39;) ; mv $i $a ; done
  
  # send an array of GSMR
  tmp_command=&quot;gcta64 --mbfile gsmr_ref_data.txt --gsmr-file gsmr_exposure.txt x{TASK_ID}  --gsmr-direction 0 --out gsmr_aftermtcojo${myRiskFact}_result_vitaminDXiaEtAl_{TASK_ID}&quot;
  qsubshcom &quot;$tmp_command&quot; 1 30G mtCOJO_GSMR_array 10:00:00 &quot;-array=1-72&quot;

  # Back to main folder
  cd ..
  
done</code></pre>
<p>Then I need to clean the results in each folder and aggregate them in a summary file.</p>
<pre class="r"><code># Once it&#39;s over, concatenate the results in a unique file
for myRiskFact in  $(ls mtcojo_*_result_vitaminD.mtcojo.cma | sed &#39;s/.*cojo_//&#39; | sed &#39;s/_.*//&#39;) ; do
  cd /shares/compbio/Group-Wray/YanHoltz/VITAMIND_XIA_ET_AL/8_MTCOJO/GSMR_${myRiskFact}
  cat gsmr_aftermtcojo*gsmr | head -1 &gt; gsmr_aftermtcojo${myRiskFact}_result_vitaminDXiaEtAl.gsmr
  cat gsmr_aftermtcojo*gsmr | grep -v &quot;Exposure&quot; &gt;&gt;  gsmr_aftermtcojo${myRiskFact}_result_vitaminDXiaEtAl.gsmr
  mv gsmr_aftermtcojo${myRiskFact}_result_vitaminDXiaEtAl.gsmr ..
  cd ..
done

# Clean
rm -r GSMR*</code></pre>
<p>Transfer the results locally for further analysis.</p>
<pre class="bash"><code># Then from locally to delta
cd /Users/y.holtz/Dropbox/QBI/4_UK_BIOBANK_GWAS_PROJECT/VitaminD-GWAS/0_DATA/MTCOJO_GSMR
scp y.holtz@delta.imb.uq.edu.au:/shares/compbio/Group-Wray/YanHoltz/VITAMIND_XIA_ET_AL/8_MTCOJO/gsmr_aftermtcojo*_result_vitaminDXiaEtAl.gsmr .</code></pre>
</div>
<div id="result" class="section level1 tabset">
<h1>Result</h1>
<hr />
<p>The first step is to merge all the GSMR results. I’m gonna put together:</p>
<ul>
<li>the initial gsmr result done with the Vitamin D GWAS summary statistics</li>
<li>all the gsmr results using the GWAS corrected by one trait with mt-COJO</li>
<li>the gsmr results using the GWAS corrected by all the traits with mt-COJO</li>
</ul>
<pre class="r"><code># Show the information of GSMR withour mtCOJO
initial &lt;- read.table(&quot;0_DATA/gsmr_result_vitaminDXiaEtAl.gsmr&quot;, header = T) %&gt;%
    dplyr::select(Outcome, bxy, p, se)
colnames(initial) &lt;- c(&quot;Outcome&quot;, &quot;bxy&quot;, &quot;p&quot;, &quot;se&quot;)

# Read the new gsmr results
all_new &lt;- list.files(&quot;0_DATA/MTCOJO_GSMR&quot;)
gsmr &lt;- initial %&gt;% mutate(name=&quot;initial&quot;)
for(i in all_new){
  new &lt;- read.table(paste0(&quot;0_DATA/MTCOJO_GSMR/&quot;,i), header = T) %&gt;% dplyr::select(Outcome, bxy, p, se)
  new$name &lt;-  i %&gt;% gsub(&quot;gsmr_aftermtcojo&quot;, &quot;&quot;, .) %&gt;% gsub(&quot;_result_vitaminDXiaEtAl.gsmr&quot;, &quot;&quot;, .)
  gsmr &lt;- rbind(gsmr, new)
}

# Correct the &quot; &quot;
gsmr$Outcome &lt;- gsub(&quot;_&quot;, &quot; &quot;, gsmr$Outcome)

# The risk factors are encoded, I need to translate them
gsmr$name &lt;- recode(gsmr$name, ALL=&quot;All risk factors&quot;, EY=&quot;Education years&quot;, height=&quot;Height&quot;, IPAQ=&quot;Outdoor sport activity (IPAQ)&quot;, SMOKE=&quot;Smoking&quot;, TR=&quot;Tanning Response&quot;, WHR=&quot;Waist Hip Ratio corrected by BMI&quot;)

# Read the meaning of files:
meaning=read.xls(&quot;0_DATA/list_of_traits_GSMR.xlsx&quot;, header=T)
primary &lt;- gsmr %&gt;% filter( Outcome %in% meaning$Trait[ which(meaning$Group==&quot;Primary&quot;) ] )
secondary &lt;- gsmr %&gt;% filter( Outcome %in% meaning$Trait[ which(meaning$Group!=&quot;Secondary&quot;) ] )

# Compute thresholds
thres_primary &lt;- 0.05 / nrow(primary)
thres_secondary &lt;- 0.05 / nrow(secondary)</code></pre>
<p>Now we can visualize the results:</p>
<ul>
<li>Each row represents the gsmr results related to a primary outcome</li>
<li>The X axis represents the effect of vitamin-D on the outcome. If this effect is <em>positive</em>, then more vitamin-D leads to more of this outcome.</li>
<li>Big red circles represent the initial gsmr (see the gsmr tab for more details)</li>
<li>The small grey circles represent the gsmr result using vitamin-D GWAS sumstat corrected by a risk factor using mt-COJO.</li>
</ul>
<pre class="r"><code># Plot
p &lt;- primary %&gt;%
  arrange(bxy) %&gt;%
  filter(!is.na(bxy) ) %&gt;% 
  mutate(Outcome=factor(Outcome, unique(Outcome))) %&gt;%
  mutate(significance=ifelse(p&lt;thres_primary, &quot;signif multi.testing&quot;, ifelse(p&lt;0.05, &quot;signif&quot;, &quot;nonSignif&quot;) ) ) %&gt;%
  mutate( mycolor = ifelse(name==&quot;initial&quot;,&quot;init&quot;,&quot;corec&quot;) ) %&gt;%
  mutate(mytext = name) %&gt;%
  ggplot( aes(x=Outcome, y=bxy, color=mycolor, size=mycolor, text=mytext)) +
    geom_hline( yintercept=0 ) +
    geom_point() +
    scale_color_manual(values=c(&quot;grey&quot;, &quot;red&quot;)) +
    scale_size_manual(values=c(1,2)) +
    coord_flip() +
    theme_ipsum() +
    theme( 
      panel.grid.major.y = element_line(size=0.1),
      legend.position=&quot;none&quot;
    ) +
    ylab(&quot;Bxy (Effect size)&quot;) +
    xlab(&quot;&quot;) 

ggplotly(p, tooltip=&quot;text&quot;)</code></pre>
<div id="6f816b4876a" style="width:864px;height:768px;" class="plotly html-widget"></div>
<script type="application/json" data-for="6f816b4876a">{"x":{"data":[{"x":[0,0],"y":[-0.504033,0.216379],"text":"","type":"scatter","mode":"lines","line":{"width":1.88976377952756,"color":"rgba(0,0,0,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-0.471287,-0.46985,-0.454106,-0.448996,-0.448996,-0.448996,-0.448996,-0.277093,-0.270037,-0.268125,-0.265207,-0.265207,-0.265207,-0.265207,-0.198607,-0.198229,-0.194681,-0.194681,-0.194681,-0.194681,-0.184092,-0.184092,-0.184092,-0.184092,-0.178048,-0.169701,-0.169701,-0.169701,-0.169701,-0.157335,-0.156764,-0.156573,-0.152897,-0.143023,-0.143023,-0.143023,-0.143023,-0.142719,-0.142719,-0.142719,-0.142719,-0.140781,-0.139598,-0.138892,-0.132603,-0.132057,-0.128629,-0.128205,-0.127228,-0.120898,-0.120643,-0.119132,-0.117248,-0.117248,-0.117248,-0.117248,-0.115219,-0.115219,-0.115219,-0.115219,-0.108737,-0.10638,-0.106079,-0.0914892,-0.0903309,-0.090325,-0.0862347,-0.0862347,-0.0862347,-0.0862347,-0.0818216,-0.0818216,-0.0818216,-0.0818216,-0.0817046,-0.0817046,-0.0817046,-0.0817046,-0.078836,-0.0778506,-0.0757938,-0.0753773,-0.0750249,-0.0727205,-0.0695121,-0.0688371,-0.0684785,-0.0673043,-0.0673043,-0.0673043,-0.0673043,-0.0610945,-0.0610945,-0.0610945,-0.0610945,-0.0478488,-0.0451323,-0.0435835,-0.0433942,-0.0433909,-0.0411652,-0.0411652,-0.0411652,-0.0411652,-0.0388667,-0.0384767,-0.0308,-0.0292633,-0.0292633,-0.0292633,-0.0292633,-0.0281917,-0.028145,-0.0259634,-0.024106,-0.0233021,-0.0229713,-0.0229713,-0.0229713,-0.0229713,-0.0229339,-0.0174335,-0.0174335,-0.0174335,-0.0174335,-0.0153697,-0.0153697,-0.0153697,-0.0153697,-0.014354,-0.0133888,-0.0127987,-0.00344078,-0.00344078,-0.00344078,-0.00344078,-0.00294016,-0.00282019,-0.00282019,-0.00282019,-0.00282019,-0.00234698,-0.00230617,0.000937481,0.000937481,0.000937481,0.000937481,0.00283774,0.00387763,0.00427858,0.00482061,0.00622025,0.00622025,0.00622025,0.00622025,0.00775649,0.00865116,0.0102203,0.0118665,0.0118665,0.0142019,0.0143748,0.0145178,0.0147402,0.0147402,0.0252605,0.0255615,0.0255615,0.0255615,0.0259676,0.0261525,0.026457,0.0332331,0.0332331,0.0345727,0.0355575,0.0355575,0.0366843,0.0378991,0.0574842,0.0591033,0.0703102,0.178591,0.1788,0.183467,0.183467,0.183467,0.183467,0.183633],"y":[1,1,1,1,1,1,1,2,2,2,2,2,2,2,3,3,3,3,3,3,4,4,4,4,3,5,5,5,5,6,6,6,4,7,7,7,7,6,6,6,6,4,4,5,8,8,7,5,8,7,7,5,9,9,9,9,8,8,8,8,10,10,10,9,9,9,10,10,10,10,11,11,11,11,12,12,12,12,12,11,12,11,11,12,13,13,13,13,13,13,13,14,14,14,14,15,16,16,15,15,16,16,16,16,16,17,18,15,15,15,15,18,18,14,14,19,18,18,18,18,14,20,20,20,20,21,21,21,21,17,17,20,19,19,19,19,22,22,22,22,22,22,22,23,23,23,23,19,20,19,20,24,24,24,24,23,23,23,25,25,24,24,24,25,25,17,17,17,17,26,26,26,26,26,25,26,26,25,25,21,21,21,27,27,27,27,27,27,27],"text":["Waist Hip Ratio corrected by BMI","Education years","Tanning Response","All risk factors","Height","Outdoor sport activity (IPAQ)","Smoking","Education years","Waist Hip Ratio corrected by BMI","Tanning Response","All risk factors","Height","Outdoor sport activity (IPAQ)","Smoking","Waist Hip Ratio corrected by BMI","Education years","All risk factors","Height","Outdoor sport activity (IPAQ)","Smoking","All risk factors","Height","Outdoor sport activity (IPAQ)","Smoking","Tanning Response","All risk factors","Height","Outdoor sport activity (IPAQ)","Smoking","Education years","Tanning Response","Waist Hip Ratio corrected by BMI","Tanning Response","All risk factors","Height","Outdoor sport activity (IPAQ)","Smoking","All risk factors","Height","Outdoor sport activity (IPAQ)","Smoking","Education years","Waist Hip Ratio corrected by BMI","Education years","Education years","Tanning Response","Tanning Response","Tanning Response","Waist Hip Ratio corrected by BMI","Education years","Waist Hip Ratio corrected by BMI","Waist Hip Ratio corrected by BMI","All risk factors","Height","Outdoor sport activity (IPAQ)","Smoking","All risk factors","Height","Outdoor sport activity (IPAQ)","Smoking","Tanning Response","Education years","Waist Hip Ratio corrected by BMI","Education years","Waist Hip Ratio corrected by BMI","Tanning Response","All risk factors","Height","Outdoor sport activity (IPAQ)","Smoking","All risk factors","Height","Outdoor sport activity (IPAQ)","Smoking","All risk factors","Height","Outdoor sport activity (IPAQ)","Smoking","Waist Hip Ratio corrected by BMI","Education years","Education years","Waist Hip Ratio corrected by BMI","Tanning Response","Tanning Response","Waist Hip Ratio corrected by BMI","Education years","Tanning Response","All risk factors","Height","Outdoor sport activity (IPAQ)","Smoking","All risk factors","Height","Outdoor sport activity (IPAQ)","Smoking","Tanning Response","Waist Hip Ratio corrected by BMI","Education years","Waist Hip Ratio corrected by BMI","Education years","All risk factors","Height","Outdoor sport activity (IPAQ)","Smoking","Tanning Response","Tanning Response","Tanning Response","All risk factors","Height","Outdoor sport activity (IPAQ)","Smoking","Waist Hip Ratio corrected by BMI","Education years","Education years","Waist Hip Ratio corrected by BMI","Tanning Response","All risk factors","Height","Outdoor sport activity (IPAQ)","Smoking","Tanning Response","All risk factors","Height","Outdoor sport activity (IPAQ)","Smoking","All risk factors","Height","Outdoor sport activity (IPAQ)","Smoking","Education years","Waist Hip Ratio corrected by BMI","Education years","All risk factors","Height","Outdoor sport activity (IPAQ)","Smoking","Waist Hip Ratio corrected by BMI","All risk factors","Height","Outdoor sport activity (IPAQ)","Smoking","Tanning Response","Education years","All risk factors","Height","Outdoor sport activity (IPAQ)","Smoking","Education years","Waist Hip Ratio corrected by BMI","Waist Hip Ratio corrected by BMI","Tanning Response","All risk factors","Height","Outdoor sport activity (IPAQ)","Smoking","Education years","Waist Hip Ratio corrected by BMI","Tanning Response","All risk factors","Height","Education years","Tanning Response","Waist Hip Ratio corrected by BMI","Outdoor sport activity (IPAQ)","Smoking","All risk factors","Height","Outdoor sport activity (IPAQ)","Smoking","Tanning Response","Waist Hip Ratio corrected by BMI","Education years","Outdoor sport activity (IPAQ)","Smoking","Waist Hip Ratio corrected by BMI","All risk factors","Height","Education years","Tanning Response","Education years","Waist Hip Ratio corrected by BMI","Tanning Response","Education years","Waist Hip Ratio corrected by BMI","All risk factors","Height","Outdoor sport activity (IPAQ)","Smoking","Tanning Response"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(190,190,190,1)","opacity":1,"size":3.77952755905512,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(190,190,190,1)"}},"hoveron":"points","name":"corec","legendgroup":"corec","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[-0.471287,-0.270037,-0.198607,-0.156573,-0.139598,-0.127228,-0.120643,-0.119132,-0.106079,-0.0903309,-0.078836,-0.0753773,-0.0695121,-0.0451323,-0.0433942,-0.0281917,-0.024106,-0.0133888,-0.00294016,0.00387763,0.00427858,0.00865116,0.0145178,0.0262072,0.0365254,0.0591033,0.1788],"y":[1,2,3,6,4,8,7,5,10,9,12,11,13,16,15,18,14,17,22,20,19,23,24,26,25,21,27],"text":"initial","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(255,0,0,1)","opacity":1,"size":7.55905511811024,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(255,0,0,1)"}},"hoveron":"points","name":"init","legendgroup":"init","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null}],"layout":{"margin":{"t":60.8318804483188,"r":39.8505603985056,"b":75.8821087588211,"l":249.896222498962},"font":{"color":"rgba(0,0,0,1)","family":"Arial Narrow","size":15.2760481527605},"xaxis":{"domain":[0,1],"type":"linear","autorange":false,"range":[-0.504033,0.216379],"tickmode":"array","ticktext":["-0.4","-0.2","0.0","0.2"],"tickvals":[-0.4,-0.2,0,0.2],"categoryorder":"array","categoryarray":["-0.4","-0.2","0.0","0.2"],"nticks":null,"ticks":"","tickcolor":null,"ticklen":3.81901203819012,"tickwidth":0,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"Arial Narrow","size":15.2760481527605},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":true,"gridcolor":"rgba(204,204,204,1)","gridwidth":0.265670402656704,"zeroline":false,"anchor":"y","title":"Bxy (Effect size)","titlefont":{"color":"rgba(0,0,0,1)","family":"Arial Narrow","size":11.9551681195517},"hoverformat":".2f"},"yaxis":{"domain":[0,1],"type":"linear","autorange":false,"range":[0.4,27.6],"tickmode":"array","ticktext":["Alzheimer's Disease","Autism Spectrum Disorder","Major Depressive Disorder","psychiatric disorder","Type II diabetes","Dementia - Alzheimer","Major depressive disorder","Parkinson's Disease","Coronary Artery Disease","Rheumatoid Arthritis","Schizophrenia","Dyslipidemia","Mean Corpuscular Hemoglobin","Asthma","Platelet ditribution width","Stroke","osteoporosis","Mean platelet volume ","Cancer","Red blood cell count","Bipolar Disorder","Mean corpuscular volume","Hematocrit","Disease count","Depressive Symptoms","Neuroticism","Neutrophil"],"tickvals":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27],"categoryorder":"array","categoryarray":["Alzheimer's Disease","Autism Spectrum Disorder","Major Depressive Disorder","psychiatric disorder","Type II diabetes","Dementia - Alzheimer","Major depressive disorder","Parkinson's Disease","Coronary Artery Disease","Rheumatoid Arthritis","Schizophrenia","Dyslipidemia","Mean Corpuscular Hemoglobin","Asthma","Platelet ditribution width","Stroke","osteoporosis","Mean platelet volume ","Cancer","Red blood cell count","Bipolar Disorder","Mean corpuscular volume","Hematocrit","Disease count","Depressive Symptoms","Neuroticism","Neutrophil"],"nticks":null,"ticks":"","tickcolor":null,"ticklen":3.81901203819012,"tickwidth":0,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"Arial Narrow","size":15.2760481527605},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":true,"gridcolor":"rgba(204,204,204,1)","gridwidth":0.132835201328352,"zeroline":false,"anchor":"x","title":"","titlefont":{"color":"rgba(0,0,0,1)","family":"Arial Narrow","size":11.9551681195517},"hoverformat":".2f"},"shapes":[{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":1}],"showlegend":false,"legend":{"bgcolor":null,"bordercolor":null,"borderwidth":0,"font":{"color":"rgba(0,0,0,1)","family":"Arial Narrow","size":12.2208385222084}},"hovermode":"closest","barmode":"relative"},"config":{"doubleClick":"reset","modeBarButtonsToAdd":[{"name":"Collaborate","icon":{"width":1000,"ascent":500,"descent":-50,"path":"M487 375c7-10 9-23 5-36l-79-259c-3-12-11-23-22-31-11-8-22-12-35-12l-263 0c-15 0-29 5-43 15-13 10-23 23-28 37-5 13-5 25-1 37 0 0 0 3 1 7 1 5 1 8 1 11 0 2 0 4-1 6 0 3-1 5-1 6 1 2 2 4 3 6 1 2 2 4 4 6 2 3 4 5 5 7 5 7 9 16 13 26 4 10 7 19 9 26 0 2 0 5 0 9-1 4-1 6 0 8 0 2 2 5 4 8 3 3 5 5 5 7 4 6 8 15 12 26 4 11 7 19 7 26 1 1 0 4 0 9-1 4-1 7 0 8 1 2 3 5 6 8 4 4 6 6 6 7 4 5 8 13 13 24 4 11 7 20 7 28 1 1 0 4 0 7-1 3-1 6-1 7 0 2 1 4 3 6 1 1 3 4 5 6 2 3 3 5 5 6 1 2 3 5 4 9 2 3 3 7 5 10 1 3 2 6 4 10 2 4 4 7 6 9 2 3 4 5 7 7 3 2 7 3 11 3 3 0 8 0 13-1l0-1c7 2 12 2 14 2l218 0c14 0 25-5 32-16 8-10 10-23 6-37l-79-259c-7-22-13-37-20-43-7-7-19-10-37-10l-248 0c-5 0-9-2-11-5-2-3-2-7 0-12 4-13 18-20 41-20l264 0c5 0 10 2 16 5 5 3 8 6 10 11l85 282c2 5 2 10 2 17 7-3 13-7 17-13z m-304 0c-1-3-1-5 0-7 1-1 3-2 6-2l174 0c2 0 4 1 7 2 2 2 4 4 5 7l6 18c0 3 0 5-1 7-1 1-3 2-6 2l-173 0c-3 0-5-1-8-2-2-2-4-4-4-7z m-24-73c-1-3-1-5 0-7 2-2 3-2 6-2l174 0c2 0 5 0 7 2 3 2 4 4 5 7l6 18c1 2 0 5-1 6-1 2-3 3-5 3l-174 0c-3 0-5-1-7-3-3-1-4-4-5-6z"},"click":"function(gd) { \n        // is this being viewed in RStudio?\n        if (location.search == '?viewer_pane=1') {\n          alert('To learn about plotly for collaboration, visit:\\n https://cpsievert.github.io/plotly_book/plot-ly-for-collaboration.html');\n        } else {\n          window.open('https://cpsievert.github.io/plotly_book/plot-ly-for-collaboration.html', '_blank');\n        }\n      }"}],"cloud":false},"source":"A","attrs":{"6f8116a5b926":{"yintercept":{},"type":"scatter"},"6f815a1e7967":{"x":{},"y":{},"colour":{},"size":{},"text":{}}},"cur_data":"6f8116a5b926","visdat":{"6f8116a5b926":["function (y) ","x"],"6f815a1e7967":["function (y) ","x"]},"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.2,"selected":{"opacity":1}},"base_url":"https://plot.ly"},"evals":["config.modeBarButtonsToAdd.0.click"],"jsHooks":{"render":[{"code":"function(el, x) { var ctConfig = crosstalk.var('plotlyCrosstalkOpts').set({\"on\":\"plotly_click\",\"persistent\":false,\"dynamic\":false,\"selectize\":false,\"opacityDim\":0.2,\"selected\":{\"opacity\":1}}); }","data":null}]}}</script>
<p><em>Note</em>: grey dots tend to overlap, but all the risk factor corrections are always present, no missing data.</p>
<p><em>Note</em>: the figure is interactive: zoom in / hover dots for more info</p>
</div>
<div id="discussion" class="section level1">
<h1>Discussion</h1>
<hr />
<ul>
<li><p>Globally, the results after correction by mtCOJO (grey) are very close from the initial GSMR result (red). This indicates that the risk factor are not confounding effect on the role of vitamin-D on studied traits.</p></li>
<li><p>The behavior of <code>smoking</code> is interesting tough: it is always the risk factor that changes results the most. Example with bipolar disorder, osteoporosis, asthma, psychiatric disorders etc.</p></li>
<li><p>with real data I’ll have to check if the significance of effects switches when using mtCOJO</p></li>
</ul>
</div>

<br><br>
<hr />
<p style="text-align: center; font-size: 80%">A work by <a href="https://github.com/holtzy/">Yan Holtz</a></p>
<p style="text-align: center; font-size: 80%"><span style="color: #808080;"><em>Yan.holtz.data@gmail.com</em></span></p>

<!-- Add icon library -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">

<!-- Add font awesome icons -->
<p style="text-align: center;">
	<a href="https://twitter.com/r_graph_gallery?lang=en" class="fa fa-twitter"></a>
	<a href="https://www.linkedin.com/in/yan-holtz-2477534a/" class="fa fa-linkedin"></a>
	<a href="https://github.com/holtzy/" class="fa fa-github"></a>
</p>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
